gadgets/trace_open/trace_open_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"io/fs"
+	"testing"
+
+	"github.com/inspektor-gadget/inspektor-gadget/integration"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/command"
+	igrunner "github.com/inspektor-gadget/inspektor-gadget/pkg/testing/ig"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/match"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+	"github.com/stretchr/testify/require"
+)
+
+type Event struct {
+	eventtypes.CommonData
+
+	// Type indicates the kind of this event
+	Type string `json:"type"`
+
+	// Message when Type is ERR, WARN, DEBUG or INFO
+	Message string `json:"message,omitempty"`
+}
+
+type traceOpenEvent struct {
+	Event
+
+	MountNsID uint64      `json:"mountnsid"`
+	Pid       uint32      `json:"pid"`
+	Uid       uint32      `json:"uid"`
+	Gid       uint32      `json:"gid"`
+	Comm      string      `json:"comm"`
+	Ret       int         `json:"ret"`
+	Err       int         `json:"err"`
+	Flags     int         `json:"flags"`
+	FlagsRaw  int32       `json:"flagsRaw"`
+	Mode      int         `json:"mode"`
+	ModeRaw   fs.FileMode `json:"modeRaw"`
+	FName     string      `json:"fname"`
+	FullPath  string      `json:"fullPath"`
+}
+
+func TestTraceOpen(t *testing.T) {
+	cn := "test-trace-open"
+	containerFactory, err := integration.NewContainerFactory("docker")
+	require.NoError(t, err, "new container factory")
+
+	ig := igrunner.New(
+		igrunner.WithPath("ig"),
+		igrunner.WithImage("ghcr.io/inspektor-gadget/gadget/trace_open:latest"),
```

Comment:
I wonder if igrunner should also provide a `WithImageBuildFromPath($GADGET_DIR)`

Then, the test can first build the image with `ig image build -t $TEMP_NAME $GADGET_DIR` and then use the gadget previously built.

The test can get `$GADGET_DIR` with something like:
```
  _, filename, _, _ := runtime.Caller(0)
  gadgetDir := path.Dir(filename)
```

Maybe not in this PR.

Reply:
Yep, I would suggest to manage it in another PR.

Reply:
Actually, now that I reviewed the CI part, I'm not sure about it anymore.

I realized the `image-based-gadgets-test` job is building all the gadgets, but `build-and-push-gadgets` job is also building and pushing them.

In order to verify the push and pull features, `image-based-gadgets-test` should use the images built and pushed by `build-and-push-gadgets`. However, it makes sense only for IG's CI because we want to test the push/pull features. On the contrary, in the [gadget-template](https://github.com/inspektor-gadget/gadget-template)'s CI, we should apply @alban's suggestion and build+test things locally.